packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -6,12 +6,17 @@ import 'dart:async';
 import 'dart:io';
 import 'app_store_connection.dart';
 import 'google_play_connection.dart';
+import 'product_details.dart';
 
 /// Basic generic API for making in app purchases across multiple platforms.
 abstract class InAppPurchaseConnection {
   /// Returns true if the payment platform is ready and available.
   Future<bool> isAvailable();
 
+  /// Query the product details list.
```

Comment:
nit: I don't think we're really querying from a pre-existing list of product details here. It's more like we're querying for product details that match the given set of identifiers.